src/mlpack/core/tree/binary_space_tree/binary_space_tree_impl.hpp

```diff
@@ -1101,7 +1101,7 @@ void BinarySpaceTree<MetricType, StatisticType, MatType, BoundType, SplitType>::
     ar & CEREAL_POINTER(left);
   if (hasRight)
     ar & CEREAL_POINTER(right);
-  if (hasParent)
+  if (!hasParent)
     ar & CEREAL_POINTER(dataset);
 
   if (Archive::is_loading::value)
```

Comment:
@rcurtin This is what I have in mind If you think this is good, I will apply it to all the trees?

Reply:
@rcurtin would it be possible to include `cereal` headers in Jenkins?

Reply:
Done. :+1:

Reply:
Also, yes, I believe that code for `BinarySpaceTree` serialization will work correctly (you can see if it passes the tests, they are pretty comprehensive).